src/main/java/seedu/address/model/modulelesson/ModuleLesson.java

```diff
@@ -71,6 +71,17 @@ public boolean isSameModuleLesson(ModuleLesson otherModuleLesson) {
                 && otherModuleLesson.getModuleCode().getLessonCodes().equals(getModuleCode().getLessonCodes());
     }
 
+    /**
+     * Returns true if the {@code otherModuleLesson} clashes with this moduleLesson.
+     */
+    public boolean clashesWith(ModuleLesson otherModuleLesson) {
+        if (getDay().equals(otherModuleLesson.getDay())) {
+            return getLessonStartTime().value.isBefore(otherModuleLesson.getLessonEndTime().value)
+                    && otherModuleLesson.getLessonStartTime().value.isBefore(getLessonEndTime().value);
+        }
+        return false;
+    }
```

Comment:
I think the check should be || instead of &&?
```suggestion
    public boolean clashesWith(ModuleLesson otherModuleLesson) {
        if (getDay().equals(otherModuleLesson.getDay())) {
            return getLessonStartTime().value.isBefore(otherModuleLesson.getLessonEndTime().value)
                    || otherModuleLesson.getLessonStartTime().value.isBefore(getLessonEndTime().value);
        }
        return false;
    }
```